Sources/GRPC/ClientConnection.swift

```diff
@@ -540,7 +540,7 @@ fileprivate extension Channel {
     do {
       let sslClientHandler = try NIOSSLClientHandler(
         context: try NIOSSLContext(configuration: configuration),
-        serverHostname: serverHostname
+        serverHostname: serverHostname.flatMap { $0.isIPAddress ? nil : $0 }
```

Comment:
I think this check is better done further up the call stack, specifically in `makeBootstrap` where we check if there is a hostname override. The rationale here is that it's easier for someone to debug if they pass in a hostname override which _is_ an IP address. It would also be useful to have a log message saying that we're not supplying a hostname override because we believe that it is an IP address.

Reply:
That makes sense. I moved that check up to `makeBootstrap`. Also added a debug log message.